GitTfs/Core/GitTfsRemote.cs

```diff
@@ -325,9 +325,12 @@ public IFetchResult FetchWithMerge(int mergeChangesetId, bool stopOnFailMergeCom
             var latestChangesetId = GetLatestChangesetId();
             if (lastChangesetIdToFetch != -1)
                 latestChangesetId = Math.Min(latestChangesetId, lastChangesetIdToFetch);
-            // TFS 2010 doesn't like when we ask for history past its last changeset.
+
             if (MaxChangesetId >= latestChangesetId)
+            {
+                Trace.WriteLine("TFS 2010 doesn't like when we ask for history past its last changeset.");
```

Comment:
Is there something that git-tfs could say here that would point you in the right direction of what to do? Or that says more explicitly that the state is unexpected. Maybe also whether things should be working or not. Showing the version mismatch might help people figure out what's happening, too.